src/realm/db.cpp

```diff
@@ -1357,6 +1358,98 @@ void DB::open(Replication& repl, const std::string& file, const DBOptions option
     do_open(file, no_create, is_backend, options); // Throws
 }
 
+void DB::create_new_history(Replication& repl)
```

Comment:
I did actually consider that approach, but I thought it would be too big an overhead to create all the state and then discard it. And the solution I present here could be kept quite simple.

Reply:
Good points, the solution here is actually not that big or complex 👍